src/query/storage/m3/cluster_resolver.go

```diff
@@ -204,6 +210,15 @@ func resolveClusterNamespacesForQueryLogicalPlan(
 			}
 		}
 
+		if !result[0].narrowing.end.IsZero() {
```

Comment:
nit: would you mind extracting this check to `narrowing` and naming it `includesMostRecentData` or smth like that?

Reply:
👍